third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -376,6 +376,171 @@ createStreamOps(const LoadToInfoMap &loadToInfo, scf::ForOp &forOp,
   return loadToStreamOp;
 }
 
+static bool isCoalesced(RankedTensorType loadType,
+                        ttg::LinearEncodingAttr llEnc) {
+  // Expect a BlockedEncoding on the load.
+  auto *ctx = loadType.getContext();
+  auto loadEnc = loadType.getEncoding();
+  auto blockedEnc = dyn_cast_or_null<ttg::BlockedEncodingAttr>(loadEnc);
+  auto shape = loadType.getShape();
+  if (!blockedEnc)
+    return false;
+
+  // Contiguous (fastest) dimension as defined by the blocked encoding.
+  const unsigned contigDim = blockedEnc.getOrder()[0];
+  const unsigned contigPerThreadBlocked =
+      blockedEnc.getSizePerThread()[contigDim];
+
+  // This is the correct way to compute vectorization instead of using
+  // getContigPerThread. However, currently global load vectorizer doesn't
+  // support vectorization that require in thread permutation (NOTE: local_load
+  // op lowering does support this!) such as: #ttg.linear<{register = [[0, 2],
+  // [0, 1]], ...}>, so we don't use largest vectorization here as well. This
+  // should be updated once vectorization in load op lowering is fixed..
+  //
+  // auto ctaLayout = ttg::getCTALayout(loadType.getEncoding());
+  // // Dummy shared layout that emulates global memory so we can use
+  // // largestVectorisation utility.
+  // auto sharedEncoding = ttg::SwizzledSharedEncodingAttr::get(
+  //     ctx, 1, 1, 1, blockedEnc.getOrder(), ctaLayout);
+  // auto sharedLL = triton::gpu::toLinearLayout(shape, sharedEncoding);
+  // auto invertedLL = ll.invertAndCompose(sharedLL).flattenOuts();
+
+  // auto [contigPerThreadLL, permutation] =
+  //     largestVectorisation(ctx, invertedLL, bitwidth, std::nullopt);
+
+  const unsigned contigPerThreadLL = llEnc.getContigPerThread()[contigDim];
```

Comment:
Nit: I'd name these as `original..` and `target..` to better reflect and contrast. `LL` is too common and applies to both..